### PR TITLE
EES-4727 tests and tidying

### PR DESCRIPTION
**Pull request by amyb-hiveit**
- Fixed the form error so it only shows when there's an error
- Checked API error handling works
- Fixed message icon size and position
- Added a page title
- Removed unneeded back link
- Fixed the send message request and set `strangle_api` to false by default
- Added more tests
- Changed the chatbot service slightly to make it easier to mock
- Removed animations (generally frowned upon in GDS-land)

### CODE REVIEW

File: chatbot-ui/components/UserInputDialog.test.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  });
+
+  it('Calls `sendMessage` when successfully submitted', async () => {
+    const handleSend = jest.fn();
+    render(
+      <UserInputDialog sendMessage={handleSend} fetching={false} error="" />,
+    );
+
+    fireEvent.change(screen.getByLabelText('What is your question?'), {
+      target: { value: 'Is the ski village on fire?' },
```
Comment (benoutram) [inline chatbot-ui/components/UserInputDialog.test.tsx:28]: 😆

File: chatbot-ui/components/UserInputDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          className="govuk-button"
+          disabled={fetching}
+          data-module="govuk-button"
+        >
+          Send
+          {fetching && (
+            <div className={styles.loadingwheel}>
+              <LoadingDots color="#000" />
+            </div>
+          )}
```
Comment (benoutram) [inline chatbot-ui/components/UserInputDialog.tsx:93]: This was something in the code before the GDS style was applied which has been broken now (or at least I can't see it). I think we can remove the LoadingDots component and its style now.

File: chatbot-ui/components/ErrorSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// TODO: Improve this component (I have only naively copied across what was already in app)
+// I don't think this needs to be a list, nor an anchor tag. Or alternatively we take an actual list of errors in the props.
```
Comment (benoutram) [inline chatbot-ui/components/ErrorSummary.tsx:2]: This code came from [Error summary](https://design-system.service.gov.uk/components/error-summary/) so I think it is correct, although clicking the link "Enter a question" in the error summary doesn't leave focus in the field. Maybe it should?

**Review by benoutram — COMMENTED** [review]
I think we could review the remaining TODO's and either action them or remove them if they are not necessary for now.

File: chatbot-ui/pages/_document.tsx [hunk 4]
```diff
@@ -34,6 +34,7 @@ export default function Document() {
           rel="apple-touch-icon"
           href="/assets/images/govuk-apple-touch-icon.png"
         />
+        <title>Chatbot prototype</title>
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply chatbot-ui/pages/_document.tsx]: Good spot, I forgot about Next's rules for it. I've moved it to `_app.tsx` now.

File: chatbot-ui/components/UserInputDialog.tsx (hunk 2 above)
Comment (amyb-hiveit) [reply chatbot-ui/components/UserInputDialog.tsx]: I've removed it now.

File: chatbot-ui/components/ErrorSummary.tsx (hunk 3 above)
Comment (amyb-hiveit) [reply chatbot-ui/components/ErrorSummary.tsx]: Yes, the code is correct. When we do the full version it should behave as it does on EES but for the prototype I think it's fine to leave it hard coded to only accept one error and to not have the correct focus behaviour. I've removed the comment.

**Review by benoutram — APPROVED** [review]

**Pull request merged**